tortoise/backends/base/schema_generator.py

```diff
@@ -144,18 +144,13 @@ def _make_hash(*args: str, length: int) -> str:
         return sha256(";".join(args).encode("utf-8")).hexdigest()[:length]
 
     def _generate_index_name(
-        self, prefix: str, model: "type[Model] | str", field_names: list[str]
+        self, prefix: str, model: type[Model] | str, field_names: list[str]
     ) -> str:
         # NOTE: for compatibility, index name should not be longer than 30
         # characters (Oracle limit).
         # That's why we slice some of the strings here.
         table_name = model if isinstance(model, str) else model._meta.db_table
-        index_name = "{}_{}_{}_{}".format(
-            prefix,
-            table_name[:11],
-            field_names[0][:7],
-            self._make_hash(table_name, *field_names, length=6),
-        )
+        index_name = f"{prefix}_{table_name[:11]}_{field_names[0][:7]}_{self._make_hash(table_name, *field_names, length=6)}"
```

Comment:
This is quite hard to read. I think it would be better to move at least the hash from the f-string:

```
h = self._make_hash(table_name, *field_names, length=6)
index_name = f"{prefix}_{table_name[:11]}_{field_names[0][:7]}_{h}"
```

For instance, Django's code style disallows having expressions in f-strings, see https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/. Personally, I think simple expressions are fine but here we have quite a lot of complexity.

Reply:
Updated.
Use template `{prefix}_{table}_{field}_{hash}` and `pk_{f}_{t}_{h}`